Solutions/Corvus.Storage.Common/Corvus/Storage/CachingStorageContextFactory.cs

```diff
@@ -130,13 +135,34 @@ public async ValueTask<TStorageContext> GetStorageContextAsync(TConfiguration co
         /// <summary>
         /// Create the context.
         /// </summary>
-        /// <param name="configuration">The container configuration.</param>
+        /// <param name="contextConfiguration">
+        /// Configuration describing the storage context.
+        /// </param>
         /// <param name="connectionOptions">Connection options (e.g., retry settings).</param>
         /// <returns>A <see cref="ValueTask"/> the produces the instance of the context.</returns>
         protected abstract ValueTask<TStorageContext> CreateContextAsync(
-            TConfiguration configuration,
+            TConfiguration contextConfiguration,
             TConnectionOptions? connectionOptions);
 
+        /// <summary>
+        /// Produces a unique cache key based on the combination of a particular storage context
+        /// that the configuration identifies, and a particular set of connection options.
+        /// </summary>
+        /// <param name="contextConfiguration">
+        /// Configuration describing the storage context.
+        /// </param>
+        /// <param name="connectionOptions">Connection options (e.g., retry settings).</param>
+        /// <returns>
+        /// A key that is unique to the combination of the storage context identified by this
+        /// configuration and the specified connection options.
+        /// </returns>
+        protected virtual string GetCacheKeyForContext(
```

Comment:
Why is this not also renamed to `GetCacheKeyForConfiguration` like the rest?

Reply:
There are three different names in use:

* `GetCacheKeyForConnectionOptions` which returns a cache key just for the connection options
* `GetCacheKeyForConfiguration` which returns a cache key just for the configuration
* `GetCacheKeyForContext` which returns a cache key incorporating everything unique to this particular request for a context, i.e., the combination of the connection options and configuration